src/state/swap/hooks.ts

```diff
@@ -109,150 +97,212 @@ export function tryParseAmount(value?: string, currency?: Currency, chainId?: nu
   parsedAmount: CurrencyAmount | undefined
   trade: Trade | undefined
   allPlatformTrades: (Trade | undefined)[] | undefined
   inputError?: number
+  loading: boolean
 }
 
 // from the current swap inputs, compute the best trade and return it.
 export function useDerivedSwapInfo(platformOverride?: RoutablePlatform): UseDerivedSwapInfoResult {
-  const dispatch = useDispatch()
   const { account, chainId } = useActiveWeb3React()
+  // Get all options for the input and output currencies
   const {
     independentField,
     typedValue,
     [Field.INPUT]: { currencyId: inputCurrencyId },
     [Field.OUTPUT]: { currencyId: outputCurrencyId },
     recipient,
   } = useSwapState()
-
+  const allowedSlippage = useUserSlippageTolerance()
+  const useMultihops = useIsMultihop()
+  const recipientLookup = useENS(recipient ?? undefined)
   const inputCurrency = useCurrency(inputCurrencyId)
   const outputCurrency = useCurrency(outputCurrencyId)
-
+  // Start by retrieveing the balances of the input and output currencies
   const relevantTokenBalances = useCurrencyBalances(account ?? undefined, [
     inputCurrency ?? undefined,
     outputCurrency ?? undefined,
   ])
-
-  const isExactIn: boolean = independentField === Field.INPUT
+  // Internal state
+  const [loading, setLoading] = useState<boolean>(false)
+  const [inputError, setInputError] = useState<number | undefined>()
+  // const [trade, setTrade] = useState<Trade>()
+  const [allPlatformTrades, setAllPlatformTrades] = useState<Trade[]>([])
+  // Computed on the fly state
+  const isExactIn = useMemo(() => independentField === Field.INPUT, [independentField])
   const parsedAmount = tryParseAmount(typedValue, (isExactIn ? inputCurrency : outputCurrency) ?? undefined, chainId)
 
-  const { trades: inTrades, loading: inLoading } = useEcoRouterExactIn(
-    isExactIn ? parsedAmount : undefined,
-    outputCurrency ?? undefined
+  // useCurrency and useToken returns a new object every time,
+  // so we need to compare the addresses as strings
+  const parsedAmountString = `${parsedAmount?.currency.address?.toString()}-${parsedAmount?.raw?.toString()}`
+  const recipientLookupComputed = `${recipientLookup.loading}-${recipientLookup?.address}-${recipientLookup?.name}`
+
+  useWhatChanged(
+    [
+      account,
+      useMultihops,
+      recipientLookupComputed,
+      chainId,
+      inputCurrency?.address,
+      outputCurrency?.address,
+      parsedAmountString,
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      relevantTokenBalances[0]?.raw.toString(),
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      relevantTokenBalances[1]?.raw.toString(),
+      allowedSlippage,
+      recipient,
+      isExactIn,
+    ],
+    `account,useMultihops,recipientLookupComputed,chainId,inputCurrency?.address,outputCurrency?.address,parsedAmountString,relevantTokenBalances[0]?.raw.toString(),relevantTokenBalances[1]?.raw.toString(),allowedSlippage,recipient,isExactIn`
   )
```

Comment:
This hook might be too much - we can remove it. However, it helps to identify what causes the excessive re-renders.